Update tasks and 🛳️  mask-generation and zero-shot-object-detection (#462)
huggingface/huggingface.js: packages/tasks/src/tasks/mask-generation/about.md

### Segmentation

Image Segmentation divides an image into segments where each pixel is mapped to an object. This task has multiple variants, such as instance segmentation, panoptic segmentation, and semantic segmentation. You can learn more about segmentation on its [task page](https://huggingface.co/tasks/image-segmentation).

## Inference

Mask generation models often work in two modes: segment everything or prompt mode. 
Below, you can see segment everything mode where many masks will be returned. 

```python
from transformers import pipeline

generator =  pipeline("mask-generation", model = "Zigeng/SlimSAM-uniform-50", points_per_batch = 64, device = "cuda")
image_url = "https://huggingface.co/ybelkada/segment-anything/resolve/main/assets/car.png"
outputs = generator(image_url, points_per_batch = 64)
outputs["masks"]
# array of multiple binary masks returned for each generated mask
```

The prompt mode takes in three types of prompts.
- **Point prompt:** The user can leave a point on the image, and a meaningful segment around the point will be returned.
- **Box prompt:** The user can draw a box on the image, and a meaningful segment within the box will be returned.

> - **Text prompt:** The user can input a text, and the objects of that type will be segmented. Note that this capability has not yet been released and has only been explored in research.

Reviewer: ```suggestion
- **Text prompt:** The user can input a text, and the objects of that type will be segmented. Note that this capability is not available in all models and has only been explored in research.
```

If no model supports it yet maybe we should not list it?

> Below you can see how to input point prompting with model inference without `pipeline`. 

Reviewer: ```suggestion
Below you can see how to use an input-point prompt. It also demonstrates direct model inference without the `pipeline` abstraction.

```

```python
from transformers import SamModel, SamProcessor
from PIL import Image
import requests 

model = SamModel.from_pretrained("Zigeng/SlimSAM-uniform-50").to("cuda")
processor = SamProcessor.from_pretrained("Zigeng/SlimSAM-uniform-50")

> raw_image = Image.open(requests.get(image_url, stream=True).raw).convert("RGB")

Reviewer: maybe repeat the url here, so the example is self-contained?

> input_points = [[[450, 600]]]

Reviewer: maybe briefly explain the shape? (bs, number_of_points, 2 (coords per point)). Also, does the processor reshape to include the batch size if we just pass a list of points?

inputs = processor(raw_image, input_points=input_points, return_tensors="pt").to("cuda")
outputs = model(**inputs)

> masks = processor.image_processor.post_process_masks(outputs.pred_masks.cpu(), inputs["original_sizes"].cpu(), inputs["reshaped_input_sizes"].cpu())

Reviewer: Is there a direct path from the processor to `post_process_masks`?

Author: sorry this is my fault I think you're right

## Inference

Mask generation models often work in two modes: segment everything or prompt mode.
The example below works in segment-everything-mode, where many masks will be returned.

```python
from transformers import pipeline

generator = pipeline("mask-generation", model="Zigeng/SlimSAM-uniform-50", points_per_batch=64, device="cuda")
image_url = "https://huggingface.co/ybelkada/segment-anything/resolve/main/assets/car.png"
outputs = generator(image_url)
outputs["masks"]
# array of multiple binary masks returned for each generated mask
```

Prompt mode takes in three types of prompts:

- **Point prompt:** The user can select a point on the image, and a meaningful segment around the point will be returned.
- **Box prompt:** The user can draw a box on the image, and a meaningful segment within the box will be returned.
- **Text prompt:** The user can input a text, and the objects of that type will be segmented. Note that this capability has not yet been released and has only been explored in research.

> Below you can see how to use an input-point prompt. It also demonstrates direct model inference without the `pipeline` abstraction. The input prompt here is a list of lists where outermost list is number of batches, inner one is number of points and innermost one is the actual location of the point.

Reviewer: ```suggestion
Below you can see how to use an input-point prompt. It also demonstrates direct model inference without the `pipeline` abstraction. The input prompt here is a nested list where the outermost list is the batch size (`1`), then the number of points (also `1` in this example), and the innermost list contains the actual coordinates of the point (`[450, 600]`).
```
